src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,81 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Entry<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Entry[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= table.length * LOAD_FACTOR) {
+            resize();
+        }
+        int index = key == null ? 0 : Math.abs(key.hashCode()) % table.length;
+        Entry<K, V> current = table[index];
+        if (current == null) {
+            table[index] = new Entry<>(key, value, null);
+            size++;
+        } else {
+            while (current != null) {
+                if ((key == null && current.key == null)
+                        || (key != null && key.equals(current.key))) {
+                    current.value = value;
+                    return;
+                }
+                if (current.next == null) {
+                    current.next = new Entry<>(key, value, null);
+                    size++;
+                    return;
+                }
+                current = current.next;
+            }
+        }
```

Comment:
use as less indentation as possible
```suggestion
            size++;
            return;
        } 
        while (current != null) {
            if ((key == null && current.key == null)
                    || (key != null && key.equals(current.key))) {
                current.value = value;
                return;
            }
            if (current.next == null) {
                current.next = new Entry<>(key, value, null);
                size++;
                return;
            }
            current = current.next;
        }
```